calcom/cal.com
# chore: Sync Services to update external tools

## What does this PR do?

Introduces the concept of Sync Services, used to sync external tools with key information of Cal.com users when they interact with the app, such as HelpScout, Close.com and SendGrid. See #3633 for more details on what's needed.

~Note that the work for this was implemented above #3709, which means it needs to be merged before this PR gets merged.~

Implements #3633.

Please be sure to also review closely related PRs in other apps. Once this PR is merged, they also need to be merged.

 - Website: https://github.com/calcom/website/pull/206
 - API: https://github.com/calcom/api/pull/155
 - Console: https://github.com/calcom/console/pull/29

## Type of change




## How should this be tested?

 - Create a new web user via website signup and see user being created in Cal.com's Sendgrid, not Close.com
 - Create a new console user via console wizard and see user being created in Cal.com's Sendgrid, not Close.com
 - Have a user create a new booking and see user's information be updated in Cal.com's Sendgrid, not Close.com (Last booking)
 - Have a user update their profile name and/or username and see user's information be updated in Cal.com's Sendgrid and Close.com (Name and Username)
 - Have a user create/update a team, a new lead with team owner as contact will be created in Close.com (contact as Point of Contact)
 - Have a user delete a team, will trigger Close.com lead deletion
 - Have a user accept/leave a team, it will be added/removed into Close.com corresponding lead
 - Have a user update a team name, it will be updated into Close.com corresponding lead
 - Have a user update their plan and see user's information be updated in Cal.com's Sendgrid, not Close.com (Plan)
 - Help Scout dedicated API to access last booking, username and plan for the ticket customer in case they exist in the system
